src/lib/flight_tasks/tasks/ManualAltitude/FlightTaskManualAltitude.cpp

```diff
@@ -289,18 +299,32 @@ void FlightTaskManualAltitude::_rotateIntoHeadingFrame(Vector2f &v)
 
 void FlightTaskManualAltitude::_updateHeadingSetpoints()
 {
-	/* Yaw-lock depends on stick input. If not locked,
-	 * yaw_sp is set to NAN.
-	 * TODO: add yawspeed to get threshold.*/
-	if (fabsf(_yawspeed_setpoint) > FLT_EPSILON) {
-		// no fixed heading when rotating around yaw by stick
-		_yaw_setpoint = NAN;
+	_isYawInput()
+	? _unlockYaw()
+	: _lockYaw();
```

Comment:
This is a very uncommon way to use the ternary operator.

Reply:
I thought the same. Especially not in the context of an assignment.

Reply:
Probably @bresch already fixed it.